Content.Server/Speech/EntitySystems/OwOAccentSystem.cs

```diff
@@ -1,41 +1,175 @@
-            " (・`ω´・)", " ;;w;;", " owo", " UwU", " >w<", " ^w^"
-        }.AsReadOnly();
-
-        private static readonly IReadOnlyDictionary<string, string> SpecialWords = new Dictionary<string, string>()
-        {
-            { "you", "wu" },
-        };
+        // Dictionary to hold words to be replaced.
+        private static Dictionary<string, string> SpecialWords = new Dictionary<string, string>();
 
         public override void Initialize()
         {
+            AddAllReplacements();
+
             SubscribeLocalEvent<OwOAccentComponent, AccentGetEvent>(OnAccent);
         }
 
         public string Accentuate(string message)
         {
+            // First replace any matched words by those registered in SpecialWords
             foreach (var (word, repl) in SpecialWords)
             {
-                message = message.Replace(word, repl);
+                //Matches any whole words (so no spaces or punctuation) and replaces it if found
+                message = Regex.Replace(message, $"\\b{word}\\b", repl);
             }
 
-            return message.Replace("!", _random.Pick(Faces))
-                .Replace("r", "w").Replace("R", "W")
-                .Replace("l", "w").Replace("L", "W");
+            // Replaces any occurrences of L's, R's, or CK's with a W, capitalized or not capitalized.
+            return message.Replace("r", "w").Replace("R", "W")
+                .Replace("l", "w").Replace("L", "W")
+                .Replace("ck", "cc").Replace("Ck", "Cc")
+                .Replace("cK", "cC").Replace("CK", "CC")
+                .Replace("tt", "dd").Replace("Tt", "Dd")
+                .Replace("tT", "dD").Replace("TT", "DD");
         }
 
         private void OnAccent(EntityUid uid, OwOAccentComponent component, AccentGetEvent args)
         {
             args.Message = Accentuate(args.Message);
         }
+
+        private static void AddReplacementSet(Dictionary<string, string> dictionary, string original, string replacement)
+        {
+            // Check if a key doesn't already exist. If it does not, add the new entry uncapitalized, First letter capitalized and ALL CAPS.
+            if (!dictionary.ContainsKey(original.ToLower()))
+            {
+                dictionary.Add(original.ToLower(), replacement.ToLower());
+                dictionary.Add(original.ToUpper(), replacement.ToUpper());
+                dictionary.Add(FirstCharToUpper(original.ToLower()), FirstCharToUpper(replacement.ToLower()));
+            }
+        }
+
+        private static void AddSpecialReplacementCase(Dictionary<string, string> dictionary, string original, string replacement)
+        {
+            // Check if a key doesn't already exist. If it does not, add the new entry directly as input in the method
+            if (!dictionary.ContainsKey(original))
+            {
+                dictionary.Add(original, replacement);
+            }
```

Comment:
You can use `TryAdd` syntax, Rider even suggests it.
```suggestion
            dictionary.TryAdd(original, replacement);
```